cli/src/query.rs

```diff
@@ -48,27 +82,61 @@ impl QueryCommand {
         }
     }
 
+    // This is a near-verbatim copy of ExportCommand::export
+    fn export(self, res: QueryResult, format: ExportFormatCommon) -> Result<(), Error> {
+        // We only add a trailing newline for JSON exports. Both YAML and TOML
+        // exporters already append a trailing newline by default.
+
+        let trailing_newline = format == ExportFormatCommon::Json;
+
+        if let Some(file) = self.output {
+            let mut file = fs::File::create(file).map_err(IOError::from)?;
+            serialize::to_writer_common(&mut file, format, &res)?;
+
+            if trailing_newline {
+                writeln!(file).map_err(IOError::from)?;
+            }
+        } else {
+            serialize::to_writer_common(std::io::stdout(), format, &res)?;
+
+            if trailing_newline {
+                println!();
+            }
+        }
+
+        Ok(())
+    }
+
     pub fn run(self, global: GlobalOptions) -> CliResult<()> {
         let mut program = self.inputs.prepare(&global)?;
 
         if self.inputs.customize_mode.field().is_none() {
             program.report(Warning::EmptyQueryPath, global.error_format);
         }
 
-        let found = program
-            .query()
-            .map(|field| {
-                query_print::write_query_result(
-                    &mut std::io::stdout(),
-                    &field,
-                    self.query_attributes(),
-                )
-                .unwrap()
-            })
-            .report_with_program(program)?;
-
-        if !found {
-            eprintln!("No metadata found for this field.")
+        if let Some(format) = self.format {
+            let query_res = program.query().map(|field| QueryResult(field));
+            if let Ok(res) = query_res {
+                self.export(res, format).report_with_program(program)?
+            } else {
```

Comment:
Nitpick: instead of duplicationg `"No metadata found for this field"`, could we define `found` outside of the `if let Some(format) = self.format {...} else {...}` and just set found to `true` in the`if` (or `false` in the else, depending on the initial value of `found`), and move the `if !found` below also after the whole if-then-else?